components/public-api-server/pkg/apiv1/workspace.go

```diff
@@ -403,6 +405,21 @@ func convertWorkspaceInstance(wsi *protocol.WorkspaceInstance, shareable bool) (
 		ports = append(ports, port)
 	}
 
+	// Calculate initial workspace folder location
+	var recentFolders []string
+	location := ""
+	if config != nil {
+		location = config.WorkspaceLocation
+		if location == "" {
+			location = config.CheckoutLocation
+		}
+	}
+	if location == "" && wsCtx != nil && wsCtx.Repository != nil {
+		location = wsCtx.Repository.Name
+
+	}
+	recentFolders = append(recentFolders, filepath.Join("/workspace", location))
```

Comment:
Looks like tests do not cover scenario for where `config.WorkspaceLocation` or `config.CheckoutLocation` is populated

Reply:
done